src/app/manage-mfa/page.tsx

```diff
@@ -0,0 +1,159 @@
+                        Use QR Code instead
+                    </button>
+                </>
+            )}
+            <button onClick={() => setStep("add")}>Reset</button>
+
+            <p>
+                Once you have set up your authentication app, verify your code
+            </p>
+            <button onClick={() => setStep("verify")}>Verify</button>
+        </>
+    );
+}
+
+function VerifyTotpScreen({
+    setStep,
+}: {
+    setStep: React.Dispatch<React.SetStateAction<AddTotpSteps>>;
+}) {
+    const { user } = useUser();
+    const [code, setCode] = React.useState("");
+
+    const verifyTotp = async (e: React.FormEvent) => {
+        e.preventDefault();
+        try {
+            await user?.verifyTOTP({ code });
+            setStep("backupcodes");
+        } catch (err) {
+            console.error(JSON.stringify(err, null, 2));
+        }
+    };
+
+    return (
+        <>
+            <h1>Verify TOTP</h1>
+            <form onSubmit={(e) => verifyTotp(e)}>
+                <label htmlFor="totp-code">
+                    Enter the code from your authentication app
+                </label>
+                <input
+                    type="text"
+                    id="totp-code"
+                    onChange={(e) => setCode(e.currentTarget.value)}
+                />
+                <button type="submit">Verify code</button>
+                <button onClick={() => setStep("add")}>Reset</button>
+            </form>
+        </>
+    );
+}
+
+function BackupCodeScreen({
+    setStep,
+}: {
+    setStep: React.Dispatch<React.SetStateAction<AddTotpSteps>>;
+}) {
+    return (
+        <>
+            <h1>Backup codes</h1>
+            <div>
+                <p>
+                    Save this list of backup codes somewhere safe in case you
+                    need to access your account in an emergency
+                </p>
+                {/* <GenerateBackupCodes /> */}
+                <button onClick={() => setStep("success")}>Finish</button>
+            </div>
+        </>
+    );
+}
+
+function SuccessScreen() {
+    return (
+        <>
+            <h1>Success!</h1>
+            <p>
+                You have successfully added TOTP MFA via an authentication
+                application.
+            </p>
+        </>
+    );
+}
+
+export default function AddMFaScreen() {
+    const [step, setStep] = React.useState<AddTotpSteps>("add");
+    const { isLoaded, user } = useUser();
+
+    if (!isLoaded) return null;
+
+    if (!user) {
+        return <p>You must be logged in to access this page</p>;
+    }
+
+    return (
+        <>
+            {step === "add" && <AddTotpScreen setStep={setStep} />}
+            {step === "verify" && <VerifyTotpScreen setStep={setStep} />}
+            {step === "backupcodes" && <BackupCodeScreen setStep={setStep} />}
+            {step === "success" && <SuccessScreen />}
+            <Link href="/manage-mfa">Manage MFA</Link>
+        </>
+    );
+}
```

Comment:
I don't think approach will work, because I don't think we can generate the MFA seeds ourselves and then somehow tell Clerk to use them, Clerk has to handle that.

Instead I think we can render a page saying something like "You must have MFA enabled on your Profile, click here to add it" and then render `RedirectToUserProfile` when they click it: https://clerk.com/docs/components/control/redirect-to-userprofile

Reply:
a major problem preventing us from using the userprofile they provide is it gives direct access to remove mfa

Reply:
yes, that's true but as long as we block access to the site w/out MFA then they have not choice but to leave it enabled.

I don't think we have a choice but to use their profile page for MFA though, do you see any API methods to add it outside the profile page?